### PR TITLE
KOGITO-3379: [DMN Designer] Multiple DRDs support - Warn users when more than one node will be cascaded deleted

### PR DESCRIPTION
**Pull request by danielzhe**
**JIRA**: [KOGITO-3379](https://issues.redhat.com/browse/KOGITO-3379)

**Business Central**: [WAR file](https://drive.google.com/file/d/1ddogQg-etL0O17Xc91IJIsCj1v6BvHNY/view?usp=sharing)

**VS Code**: [plugin](https://drive.google.com/file/d/1gt_QU0pB0WVIocQF8tJv_7zPlMAKbi9G/view?usp=sharing)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by karreiro** [comment]
Jenkins do fdb.

**Comment by karreiro** [comment]
Jenkins retest this.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/DeleteSelectionSessionCommand.java [hunk 1]
```diff
@@ -136,31 +144,55 @@ public boolean accepts(final ClientSession session) {
...
+        // Execute the commands.
+        final CommandResult<CanvasViolation> result =
+                sessionCommandManager.execute(canvasHandler,
+                                              canvasCommandFactory.delete(elements));
+
+        // Check the results.
+        if (!CommandUtils.isError(result)) {
+            callback.onSuccess();
+        } else {
+            callback.onError((V) new ClientRuntimeError("Error deleing elements [message=" +
```
Comment (jstastny-cz) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/command/impl/DeleteSelectionSessionCommand.java:191]:
```suggestion
            callback.onError((V) new ClientRuntimeError("Error deleting elements [message=" +
```

**Review by jstastny-cz — COMMENTED** [review]
Code review done, just minor thing.

Testing is still pending.

**Review by jstastny-cz — APPROVED** [review]
Works fine, manually tested in Business Central and VSCode.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants.properties [hunk 2]
```diff
@@ -1,3 +1,10 @@
+ConfirmationDialogImplView.Yes=Yes
+ConfirmationDialogImplView.No=No
+
+DeleteNodeConfirmationDialogImpl.ConfirmationDescription=All {0} nodes will be removed from all decision requirements diagrams (DRDs).
```
Comment (karreiro) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants.properties:4]: This message here ;-)

**Review by karreiro — COMMENTED** [review]
Excellent stuff, @danielzhe! Thank you. The overall looks perfect and works locally, but I believe we will need to do a minor change.

I like the idea of having this generic dialog, however I've noticed we have a DMN specific logic that we need to move to `kie-wb-common-dmn`.

Please ping me with you need some ideas about how to move it :-)

Excellent work!

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants.properties (hunk 2 above)
Comment (danielzhe) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/resources/org/kie/workbench/common/stunner/client/widgets/resources/i18n/StunnerWidgetsConstants.properties:4]: @karreiro Thank you, message changed!

**Review by karreiro — APPROVED** [review]
Thank you, @danielzhe :-)

**Comment by karreiro** [comment]
Jenkins retest this.

**Comment by karreiro** [comment]
Jenkins retest this.

**Comment by karreiro** [comment]
Jenkins retest this.

**Comment by karreiro** [comment]
Jenkins do fdb.

**Comment by karreiro** [comment]
Jenkins do fdb (@danielzhe I believe we will need to rebase the PR to get it green 😵)

**Comment by karreiro** [comment]
Jenkins do fdb.

**Comment by karreiro** [comment]
Jenkins do fdb.

**Comment by karreiro** [comment]
Jenkins retest this.

**Comment by karreiro** [comment]
@danielzhe I've noticed the broken test is this one:
```
org.kie.workbench.common.stunner.core.client.session.command.impl.DeleteSelectionSessionCommandTest.testClearSessionInvoked
```

_Seems_ related to this PR, could you please double-check it? :-)

**Comment by danielzhe** [comment]
Jenkins do fdb

**Comment by danielzhe** [comment]
Jenkins do fdb

**Comment by danielzhe** [comment]
Jenkins do fdb

**Comment by danielzhe** [comment]
Omg. Green build. Can someone of the @kiegroup/gatekeepers, merge, please?

**Pull request merged**